SU2_CFD/src/output/CFlowOutput.cpp

```diff
@@ -510,6 +549,36 @@ void CFlowOutput::SetAnalyzeSurface(const CSolver* const*solver, const CGeometry
           config->SetSurface_Species_0(iMarker_Analyze, Species);
       }
     }
+
+    if (flamelet) {
+      for (unsigned short i_var = 0; i_var < nScalars; i_var++) {
+        su2double scalar = Surface_Scalars_Total(iMarker_Analyze, i_var);
+        std::stringstream str_i_var;
+        str_i_var  << std::setw(2) << std::setfill('0') << i_var;
+        SetHistoryOutputPerSurfaceValue("SURFACE_SCALAR_" + str_i_var.str(), scalar, iMarker_Analyze);
+        Tot_Surface_Scalar[i_var] += scalar;
+        if (i_var == 0)
+          config->SetSurface_Scalar_00(iMarker_Analyze, scalar);
+        if (i_var == 1)
+          config->SetSurface_Scalar_01(iMarker_Analyze, scalar);
+        if (i_var == 2)
+          config->SetSurface_Scalar_02(iMarker_Analyze, scalar);
+        if (i_var == 3)
+          config->SetSurface_Scalar_03(iMarker_Analyze, scalar);
+        if (i_var == 4)
+          config->SetSurface_Scalar_04(iMarker_Analyze, scalar);
+        if (i_var == 5)
+          config->SetSurface_Scalar_05(iMarker_Analyze, scalar);
+        if (i_var == 6)
+          config->SetSurface_Scalar_06(iMarker_Analyze, scalar);
+        if (i_var == 7)
+          config->SetSurface_Scalar_07(iMarker_Analyze, scalar);
+        if (i_var == 8)
+          config->SetSurface_Scalar_08(iMarker_Analyze, scalar);
+        if (i_var == 9)
+          config->SetSurface_Scalar_09(iMarker_Analyze, scalar);
+      }
```

Comment:
Ok all of this can be computed with the custom outputs and custom objectives please remove.

Reply:
OK, great, I'll have to check how that works. That will definitely clean up some code.
@danielmayer Can we have a look at this at our weekly meeting?

Reply:
You can have a look at species_transport/passive_transport_validation/passive_transport.cfg

Reply:
OK, great, so this block can be removed and we can just place this below in the config file:
```
CUSTOM_OUTPUTS = 'avg_species : MassFlowAvg{SPECIES[0]}[outlet];\
                  avg_check : Function{avg_species - SURFACE_SPECIES_0}'
SCREEN_OUTPUT = INNER_ITER WALL_TIME RMS_RES LINSOL CUSTOM
```
Does that mean that we can also remove similar lines in 
CFVMFlowSolverBase::EvaluateCommonObjFunc and it will be taken care of by the CUSTOM_OBJFUNC case?

Reply:
Yes, you'll just need to adapt the [index] of the species you want.
Yes everything can be removed, and you can make objective functions that combine averages of multiple species.

Reply:
Great, the CUSTOM_OUTPUTS seem to work nicely.
The flamelet combustion model is based a lot on the species implementation, but since they are not necessarily species, I will call the flamelet variables 'SCALAR0, etc'. We could also use the actual names given in the config file.

Reply:
I'm not sure I understand, SPECIES is the name we parse for variables of species solvers (the name also identifies the solver from where the data should come)

Reply:
The flamelet variables are in general not species, so this is why I chose not to use the name SPECIES but go for SCALAR. With the flamelet solver we transport:
- progress variable and total enthalpy (always) : we do the lookup based on these 2 variables
- additional species (optional) : source terms and diffusion term are looked up using progress variable and enthalpy

We could also use the name FLAMELET, maybe more consistent. Or use SPECIES for species transport and flamelet variables transport, and in that we re-use the same code.